doc/osh-help.md

```diff
@@ -831,10 +961,125 @@ View on the web:
 
     https://www.oilshell.org/release/$VERSION/doc/
 
+<h4 id="hash">hash</h4>
+
+    hash FLAG* NAME*
+
+Avoids searching for the location of commands by keeping a hash cache
+containing the path of previously executed commands.
+
+Name:
+
+    Name of the command whose path should be remembered, so they need not to be
+    searched for subsequent invocations. `$PATH` is used for the search, and
+    previously remembered paths are discarded. If no NAME and FLAG is supplied,
+    it prints the current hash cache.
+
+Flags:
+
+    -r       Discard all remembered locations.
+<!--    -d       Discard the remembered location of each NAME.
+    -l       Display output in a format reusable as input.
+    -p PATH  Inhibit path search, PATH is used as location for NAME.
+    -t       Print the full path of one or more NAME.-->
+
+<h4 id="type">type</h4>
+
+    type FLAG* CMD
+
+Prints information about the type of command that will be run as CMD:
+an executable file, aliase, shell builtin, or function.
+
+Flags:
+
+    -f  Don't search for CMD in functions.
+    -P  Force searching for CMD only in file executables.
+    -t  Only print the type of command of CMD: 'alias', 'keyword', 'function',
+        'builtin' or 'file'.
+<!--    -a  Print all executables that can run CMD, including files, aliases,
+        builtins and functions. If used with -p, only the executable file will
+        be printed.-->
+ 
 <h3>Word Lookup</h3>
 
+<h4 id="command">command</h4>
+
+    command FLAG? CMD ARG*
+
+Executes CMD from builtin commands or executable files, ignoring shell
+functions with the same name. ARG is passed as arguments to CMD. It also
+displays information about CMD.
+
+Flags:
+
+    -v  Print a description of CMD, similar to `type`.
+<!--    -p  Use a default value for PATH that is guaranteed to find all of the
+        standard utilities.
+    -V  Print a more verbose description of CMD.-->
+
+<h4 id="builtin">builtin</h4>
+
+    builtin CMD ARG*
+
+Executes the shell builtin CMD, with arguments ARG, ignoring the existence of
+shell functions with the same name. Typically used when a shell builtin is
+reimplemented as a shell function, but the shell function needs to execute the
+builtin.
+
 ### Interactive
 
+<h4 id="alias">alias</h4>
+
+    alias CMD=STR
+
+Defines a string STR that will be used instead of CMD when you run it. If STR
+is not provided, the current alias for CMD will be shown. If run without
+parameters, it will show a list of defined alias. To remove an existing alias,
+use `unalias`.
+
+Tips:
+
+Aliases are often used to create new compounded commands that are used as a
+shortcut, or to add default options to existing commands. For example,
+alias ports='ss -tulanp' will set up an alias for the command "port" which,
+when run, will display port information. alias ls='ls --color' will make ls
```

Comment:
I would discourage use of aliases, because they cause weird parsing problems, and you can almost laways use a function:

```
ports() {
  ss -tulanp "$@"
}

ls() {
  ls --color "$@"
}

```

or in Oil:

```
proc ports {
  ss -tulanp @ARGV
}

proc ls {
  ls --color @ARGV
}
```